providers/os/connection/device/linux/device_manager.go

```diff
@@ -52,11 +53,11 @@ func (d *LinuxDeviceManager) IdentifyMountTargets(opts map[string]string) ([]*sn
 		return []*snapshot.PartitionInfo{pi}, nil
 	}
 
-	pi, err := d.identifyViaDeviceName(opts[DeviceName])
+	partitions, err := d.identifyViaDeviceName(opts[DeviceName], opts[MountAllPartitions] == "true")
```

Comment:
we should validate that device name is not empty if `MountAllPartitions` is true, we want to be explicit in what device we're targeting

Reply:
device name is validated inside `identifyViaDeviceName`

Reply:
but it can also return if the name is empty, it then does a best-guess search

if mountall is true we should explicitly validate device name is not empty

Reply:
But if device name is empty, it will return the best-guess, mount-all will be ignored:

https://github.com/mondoohq/cnquery/blob/834ade4ca2ca6413f9225a81a2860670fa142329/providers/os/connection/device/linux/device_manager.go#L133-L142